### ISSUE TITLE
Documentation links in GitLab are broken

### ISSUE THREAD

**Issue by smiklosovic** [sameersbn/docker-gitlab#726]
I am here (1) and I want to see some readme, e.g. about environment variables.

When I click on that link, it takes me to (2) but that page does not exist and results with 404 because when you notice, there are two "help/help" instead of one in (2). When I remove one "help" in the URL path like this (3) it works ok.

I am not sure if this is fault of this docker-gitlab thing or it is broken in upstream.

(1) https://my.gitlab.net/help/
(2) https://my.gitlab.net/help/help/administration/environment_variables.md
(3) https://my.gitlab.net/help/administration/environment_variables.md

**Comment by SolidNerd** [comment]
Hey, 
I tested it on some instances of mine and I can't reproduce it. I guess it's a problem of your configuration/setup. Did you use `GITLAB_RELATIVE_URL_ROOT` ? Or perhaps it's a configuration problem with your proxy.

**Comment by frol** [comment]
Just for your info, It works fine for me on our latest setup with `GITLAB_RELATIVE_URL_ROOT=/gitlab`.

**Comment by smiklosovic** [comment]
I have it just empty, I dont think `/gitlab` does a thing. If it is empty, empty should be default as it is shown in docs.

**Comment by smiklosovic** [comment]
@sameersbn 

Setting that variable to "/" solved it. This should be default. Emptiness of that variable results to mangled links as described above.

**Comment by SolidNerd** [comment]
@sameersbn It seems that it could be broken when you are not proxing GitLab. 
Here is the code:
```
if [[ -z ${GITLAB_RELATIVE_URL_ROOT} || ${GITLAB_RELATIVE_URL_ROOT} == / ]]; then # should not be set to `/`
  GITLAB_RELATIVE_URL_ROOT=
fi
```

I will investigate it on the weekend.

**Comment by sameersbn** [comment]
@SolidNerd I tested with default `docker-compose.yml` and was not able to see this issue. Are you able to reproduce?

>```
>if [[ -z ${GITLAB_RELATIVE_URL_ROOT} || ${GITLAB_RELATIVE_URL_ROOT} == / ]]; then # should not be set to `/`
>  GITLAB_RELATIVE_URL_ROOT=
>fi

This code block unsets `GITLAB_RELATIVE_URL_ROOT`, so i dont see how setting `GITLAB_RELATIVE_URL_ROOT=/` would fix the issue

**Comment by sameersbn** [comment]
I am able to reproduce this issue by adding a trailing `/` to the url, i.e. http://git.example.com/help/. It has nothing to do with `GITLAB_RELATIVE_URL_ROOT`. I guess we need to check the nginx configs 😓

**Comment by SolidNerd** [comment]
@sameersbn  I came to the same result that adding a trailing `/` broke the url. I guess it's need to be fixed in `location / {...}` Perhaps a `try_files $uri $uri/ ` could fix it.

**Comment by sameersbn** [comment]
I agree, its a upstream issue. I think its not such a big issue that we should try to fix it. Lets wait for upstream to resolve it. We need to report it.

**Comment by ecrazor1911** [comment]
Similar issue: #703 

https://domain/help/ci%2Fquick_start/README

Maybe the problem is a escaped slash character.

**Comment by ecrazor1911** [comment]
@SolidNerd You use wrong link. Try this: `https://gitlab.com/help/api/README.md`

**Comment by sameersbn** [comment]
@ecrazor1911 The issue may be related. But in the newer gitlab versions all requests are forwarded to the gitlab-workhorse daemon. This is little to no configuring on the nginx end of the container, see https://github.com/sameersbn/docker-gitlab/blob/master/assets/runtime/config/nginx/gitlab.

I would try the official gitlab-ce image to see if the problem exists and report to upstream if it did.
